src/prefect/core/task.py

```diff
@@ -907,6 +907,22 @@ def outputs(self) -> Any:
             return_annotation = Any
         return return_annotation
 
+    def pipe(self, task: "Task", *args: Any, **kwargs: Any) -> "Task":
```

Comment:
Would you mind changing the argument name here to `__task` so it cannot collide with kwargs?

I'm tempted to ban positional args entirely (with `__task, *, **kwargs`) since the behavior may be confusing.

Reply:
Since prefect internally only uses kwargs (?), I'm happy to ban positional args.

Reply:
But are you happy with `task` being positional? Because otherwise the user will always have to `.pipe(task=t, arg=x)`.

Reply:
I solved this by changing the signature to `def pipe(self, task: "Task", /, **kwargs: Any) -> "Task"`, which lets us keep the neat "task" name, but forces it to be a positional argument so doesn't risk clashing with a keyword called "task". I've added a test for this too.